rollup.config.js

```diff
@@ -1,29 +1,39 @@
+import path from 'node:path'
 import ts from '@rollup/plugin-typescript'
 import dts from 'rollup-plugin-dts'
 import pkg from './package.json' with { type: 'json' }
 
-export default (async () => ([
-  {
-    input: './source/index.ts',
-    output: { file: pkg.main, format: 'es', banner: "'use client';" },
-    external: isExternal,
-    plugins: [ts({
-      tsconfig: './tsconfig.json',
-      compilerOptions: { declarationDir: './types' },
-    })],
-  },
-  {
-    input: './dist/types/index.d.ts',
-    output: { file: pkg.types, format: 'es' },
-    plugins: [dts(), rmOnWrite('./dist/types')],
-  },
-]))()
+export default (async () => {
+  const outDir = path.dirname(pkg.main)
+  const outDirTypes = `${outDir}/types`
 
-function isExternal (id) {
+  return [
+    {
+      input: './source/index.ts',
+      output: {
+        dir: outDir, // TODO: `file: pkg.main` - workaround via https://github.com/rollup/plugins/issues/1772#issuecomment-2519066903
```

Comment:
Using `dir` over `file` gets us past the bug in `@rollup-plugin-typescript@12.1.1`. Workaround is from https://github.com/rollup/plugins/issues/1772#issuecomment-2519066903